ImportError on  load_neuropod

## Bug

Get `ImportError` when loading a neuropod:
`ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory`

## To Reproduce

```
>>> from neuropod.loader import load_neuropod
>>> load_neuropod("neuropod_path")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/piero/dev/mlvenv/lib/python3.7/site-packages/neuropod/loader.py", line 211, in load_neuropod
    return NativeNeuropodExecutor(neuropod_path, **kwargs)
  File "/home/piero/dev/mlvenv/lib/python3.7/site-packages/neuropod/loader.py", line 114, in __init__
    from neuropod.neuropod_native import Neuropod as NeuropodNative
ImportError: libpython3.7m.so.1.0: cannot open shared object file: No such file or directory
```

This happens on this line in the neuropod test inside Ludwig:
https://github.com/uber/ludwig/blob/master/tests/integration_tests/test_neuropod.py#L118

You can use that whole code to reproduce the issue.
Note that that same code was working fine with version 0.1.x of neuropod (the internal version).


## Expected behavior

The model gets loaded.

## Environment

 - Neuropod Version: 0.2.0
 - OS: Ubuntu Linux 20.04
 - Language (Python, C++, Go bindings): Python
 - Python version: Python 3.7.7

So I recompiled python 3.7 with --enable-shared, reinstalled (make altinstall),, modified python.conf,  and created a new virtualenv.

```
ldd /home/piero/dev/mlvenv/lib/python3.7/site-packages/neuropod/neuropod_native.so
	linux-vdso.so.1 (0x00007ffd286e1000)
	libneuropod.so => /home/piero/dev/mlvenv/lib/python3.7/site-packages/neuropod/libneuropod.so (0x00007f9a8bdae000)
	libpython3.7m.so.1.0 => /usr/local/lib/libpython3.7m.so.1.0 (0x00007f9a8ba1d000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9a8b8ce000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f9a8b6ed000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9a8b6d2000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f9a8b6ad000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f9a8b6a7000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9a8b4b5000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f9a8c22b000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f9a8b4aa000)
	libutil.so.1 => /lib/x86_64-linux-gnu/libutil.so.1 (0x00007f9a8b4a5000)
```

Now i don-t get the error with libpython3, but I get a different one:
```
self = <neuropod.loader.NativeNeuropodExecutor object at 0x7f2ee9077390>
neuropod_path = 'neuropod', kwargs = {}
NeuropodNative = <class 'neuropod.neuropod_native.Neuropod'>

    def __init__(self, neuropod_path, **kwargs):
        """
        Load a Neuropod using the native bindings
    
        :param  neuropod_path:  The path to a neuropod package
        """
        # Load the model
        from neuropod.neuropod_native import Neuropod as NeuropodNative
    
        self.model = NeuropodNative(
>           neuropod_path, _REGISTERED_BACKENDS, use_ope=True, **kwargs
        )
E       RuntimeError: Neuropod Error: Got an exception when loading the model at neuropod: Neuropod Error: Loading the default backend for type 'python' failed. Error from dlopen: libneuropod_pythonbridge_backend.so: cannot open shared object file: No such file or directory

../../../mlvenv/lib/python3.7/site-packages/neuropod/loader.py:117: RuntimeError
```

Not sure it's worth debugging, as it is something probably specific to my configuration, but maybe if this is revealing of something Neuropod should take into account moving forward, help to keep debugging for figuring this out, your call.

--- a/source/deps/BUILD.pybind11
+++ b/source/deps/BUILD.pybind11
@@ -26,6 +26,6 @@ cc_library(
     ],
     visibility = ["//visibility:public"],
     deps = [
-        "@python_repo//:python",
+        "@python_repo//:python_hdrs",
     ],
 )

--- a/source/deps/BUILD.python
+++ b/source/deps/BUILD.python
@@ -17,11 +17,7 @@ package(
 )
 
 cc_library(
-    name = "python",
-    srcs = select({
-        "@bazel_tools//src/conditions:darwin": ["lib/libpython{PYTHON_VERSION}.dylib"],
-        "//conditions:default": glob(["lib/libpython{PYTHON_VERSION}.so.1.0", "lib/libpython{PYTHON_VERSION}m.so.1.0"]),
-    }),
+    name = "python_hdrs",
     hdrs = glob([
         "include/python{PYTHON_VERSION}/**/*.h",
         "include/python{PYTHON_VERSION}m/**/*.h",
@@ -31,5 +27,15 @@ cc_library(
         "include/python{PYTHON_VERSION}m",
     ],
     defines = ["PYTHON_VERSION={PYTHON_VERSION}"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "python",
+    srcs = select({
+        "@bazel_tools//src/conditions:darwin": ["lib/libpython{PYTHON_VERSION}.dylib"],
+        "//conditions:default": glob(["lib/libpython{PYTHON_VERSION}.so.1.0", "lib/libpython{PYTHON_VERSION}m.so.1.0"]),
+    }),
+    deps = [":python_hdrs"],
     visibility = ["//visibility:public"],
 )

--- a/source/neuropod/backends/python_bridge/BUILD
+++ b/source/neuropod/backends/python_bridge/BUILD
@@ -50,6 +50,7 @@ cc_library(
         "//neuropod/bindings",
         "//neuropod/core",
         "//neuropod/internal",
+        "@python_repo//:python",
     ],
     alwayslink = True,
 )

